### ISSUE TITLE
didChange event handling fails with out of range error after which every format request fails

### ISSUE THREAD

**Issue by ramya-rao-a** [sourcegraph/go-langserver#277]
The below was observed when using the language server via the Go extension to VS Code.

Below is a simple program of 12 lines. All I did was add tabs before the print statement followed by file save a couple of times. Soon there is an error when handling the `textDocument/didChange` event. After this every file save that triggers formatting ends up generating the below formatting error.

<img width="951" alt="screen shot 2018-05-10 at 9 05 10 pm" src="https://user-images.githubusercontent.com/16890566/39906216-25e81116-5496-11e8-89d3-f92e9a37dee2.png">

Weirdly, it says it found `}` in line 12, but `}` is in line 11

**Comment by keegancsmith** [comment]
yeah I think there are some issues around how we interpret ranges in go-langserver. I'll take a look soon.

By put it behind a feature flag, do you mean advertising full sync vs incremental sync?

**Comment by ramya-rao-a** [comment]
Never mind the feature flag, I figured out a way to intercept the calls going to the language server from the Go extension. So what I have done is disabled the formatting feature from the language server. User can opt-in to use the formatting feature from the language server explicitly in VS Code now.

Well, by "now" I mean the next update
